src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -153,21 +153,28 @@
                                         {:chains   session-networks
                                          :methods  constants/wallet-connect-supported-methods
                                          :events   constants/wallet-connect-supported-events
-                                         :accounts accounts}})]
-     {:fx [[:effects.wallet-connect/approve-session
-            {:web3-wallet          web3-wallet
-             :proposal             current-proposal
-             :supported-namespaces supported-namespaces
-             :on-success           (fn [approved-session]
-                                     (log/info "Wallet Connect session approved")
-                                     (rf/dispatch [:wallet-connect/reset-current-session-proposal])
-                                     (rf/dispatch [:wallet-connect/persist-session approved-session]))
-             :on-fail              (fn [error]
-                                     (log/error "Wallet Connect session approval failed"
-                                                {:error error
-                                                 :event :wallet-connect/approve-session})
-                                     (rf/dispatch
-                                      [:wallet-connect/reset-current-session-proposal]))}]
+                                         :accounts accounts}})
+         expiry               (get-in current-proposal [:params :expiryTimestamp])]
+     {:fx [(if (wc-utils/timestamp-expired? expiry)
+             [:dispatch
+              [:toasts/upsert
+               {:id   :new-wallet-account-created
+                :type :positive
+                :text (i18n/label :t/wallet-connect-proposal-expired)}]]
+             [:effects.wallet-connect/approve-session
+              {:web3-wallet          web3-wallet
+               :proposal             current-proposal
+               :supported-namespaces supported-namespaces
+               :on-success           (fn [approved-session]
+                                       (log/info "Wallet Connect session approved")
+                                       (rf/dispatch [:wallet-connect/reset-current-session-proposal])
+                                       (rf/dispatch [:wallet-connect/persist-session approved-session]))
+               :on-fail              (fn [error]
+                                       (log/error "Wallet Connect session approval failed"
+                                                  {:error error
+                                                   :event :wallet-connect/approve-session})
```

Comment:
Maybe you like this, maybe not, but we can get the event ID out of the coeffects, as we do here:

https://github.com/status-im/status-mobile/blob/e4639c153f4ff499fd4aa535fd92ad4291e6644e/src/status_im/contexts/communities/actions/addresses_for_permissions/events.cljs#L9